### PR TITLE
Fix ascii range check

### PR DESCRIPTION
**Pull request by benaadams**
cc @ahsonkhan @tarekgh, @jkotas

### CODE REVIEW

File: src/mscorlib/shared/System/Globalization/CompareInfo.cs [hunk 1]
```diff
@@ -566,7 +566,7 @@ internal static unsafe int CompareOrdinalIgnoreCase(ReadOnlySpan<char> strA, Rea
                 char* b = bp;
 
                 // in InvariantMode we support all range and not only the ascii characters.
-                char maxChar = (char) (GlobalizationMode.Invariant ? 0xFFFF : 0x80);
+                char maxChar = (char) (GlobalizationMode.Invariant ? 0xFFFF : 0x7f);
```
Comment (benaadams) [inline src/mscorlib/shared/System/Globalization/CompareInfo.cs]: Is `GlobalizationMode.Invariant` initrinsicish to the Jit? (like readonly static)

Comment (tarekgh) [reply src/mscorlib/shared/System/Globalization/CompareInfo.cs]: should use _invariantMode and not GlobalizationMode.Invariant

File: src/mscorlib/shared/System/Globalization/CompareInfo.cs [hunk 2]
```diff
@@ -74,9 +74,9 @@ public partial class CompareInfo : IDeserializationCallback
         [OptionalField(VersionAdded = 3)]
         private SortVersion m_SortVersion; // Do not rename (binary serialization)
 
-        // _invariantMode is defined for the perf reason as accessing the instance field is faster than access the static property GlobalizationMode.Invariant
+        // s_invariantMode is defined for the perf reason as accessing the field is faster than access the static property GlobalizationMode.Invariant
         [NonSerialized]
-        private readonly bool _invariantMode = GlobalizationMode.Invariant;
```
Comment (jkotas) [inline src/mscorlib/shared/System/Globalization/CompareInfo.cs]: GlobalizationMode.Invariat is a readonly static already. We do not need to duplicate it.

Comment (jkotas) [reply src/mscorlib/shared/System/Globalization/CompareInfo.cs]: This is cached in a instance field because instance field access is faster (in crossgened code in particular - and CoreLib is pretty much always crossgened).

File: src/mscorlib/shared/System/Span.NonGeneric.cs [hunk 3]
```diff
@@ -22,7 +22,10 @@ namespace System
     /// Extension methods and non-generic helpers for Span, ReadOnlySpan, Memory, and ReadOnlyMemory.
     /// </summary>
     public static class Span
-    {
+    {   
+        // s_invariantMode is defined for the perf reason as accessing the field is faster than access the static property GlobalizationMode.Invariant
+        private static readonly bool s_invariantMode = GlobalizationMode.Invariant;
```
Comment (ahsonkhan) [inline src/mscorlib/shared/System/Span.NonGeneric.cs]: See: https://github.com/dotnet/coreclr/pull/16379#discussion_r168087470

> This cache does not make sense. `GlobalizationMode.Invariant` property should have the exact same perf as you local cache.

File: src/mscorlib/shared/System/Globalization/CompareInfo.cs (hunk 1 above)
Comment (ahsonkhan) [reply src/mscorlib/shared/System/Globalization/CompareInfo.cs]: nit: Change 0x7f to 0x7F (all capital) to be consistent.

File: src/mscorlib/shared/System/Globalization/CompareInfo.cs (hunk 2 above)
Comment (tarekgh) [reply src/mscorlib/shared/System/Globalization/CompareInfo.cs]: right, we just need to replace GlobalizationMode.Invariant with _invariantMode in CompareInfo code.

**Comment by ahsonkhan** [comment]
Can you also please update this comment?https://github.com/dotnet/coreclr/blob/cb4dd6072c331d2e6a3d75d10321138020666cc4/src/mscorlib/src/System/String.Comparison.cs#L68

File: src/mscorlib/shared/System/Globalization/CompareInfo.cs (hunk 2 above)
Comment (benaadams) [reply src/mscorlib/shared/System/Globalization/CompareInfo.cs]: All the instance methods already use it; other ones are static

**Comment by benaadams** [comment]
Addressed feedback

**Comment by tarekgh** [comment]
Thanks @benaadams

**Comment by benaadams** [comment]
@dotnet-bot Tizen armel Cross Checked Innerloop Build and Test

**Comment by benaadams** [comment]
Won't let me rerun the Tizen

File: src/mscorlib/shared/System/Globalization/CompareInfo.cs [hunk 4]
```diff
@@ -566,7 +566,7 @@ internal static unsafe int CompareOrdinalIgnoreCase(ReadOnlySpan<char> strA, Rea
                 char* b = bp;
 
                 // in InvariantMode we support all range and not only the ascii characters.
-                char maxChar = (char) (GlobalizationMode.Invariant ? 0xFFFF : 0x80);
+                char maxChar = (char) (GlobalizationMode.Invariant ? 0xFFFF : 0x7F);
```
Comment (danmosemsft) [inline src/mscorlib/shared/System/Globalization/CompareInfo.cs]: Could this be computed just once in a static ctor? It's just accommodate, but this code is called a lot. (I can't see the context right now so this may be off.)

**Comment by ahsonkhan** [comment]
@dotnet-bot test Tizen armel Cross Checked Innerloop Build and Test

**Comment by ahsonkhan** [comment]
> Won't let me rerun the Tizen

Must have been some temporary issue.

**Comment by stephentoub** [comment]
Sounds like we're also missing a test in corefx?

**Comment by benaadams** [comment]
@dotnet-bot test Windows_NT x64 Checked corefx_baseline
@dotnet-bot test Ubuntu x64 Checked corefx_baseline

**Comment by benaadams** [comment]
Path failures
```
d7295605\_\fx\src\System.Runtime.Extensions\tests\System\IO\PathTests.cs(759,0): at System.IO.Tests.PathTests.GetFullPath_Windows_NotSupportedExceptionPaths(String path)
06:02:23      System.IO.Tests.PathTests.GetFullPath_Windows_UNC_Valid(expected: \"\\\\\\\\LOCALHOST\\\\share\\\\test.txt.~SS\", input: \"\\\\\\\\LOCALHOST\\\\share\\\\test.txt.~SS\") [FAIL]
06:02:23         Assert.Equal() Failure
06:02:23                                           (pos 30)
06:02:23         Expected: úúúT\share\test.txt.~SS
06:02:23         Actual:   úúúT\share\test.txt.~SS\0
06:02:23                                           (pos 30)
```

**Comment by ahsonkhan** [comment]
From Tizen armel Cross Checked Innerloop:
https://ci.dot.net/job/dotnet_coreclr/job/master/job/armel_cross_checked_tizen_innerloop_prtest/1065/consoleFull#9887747564b7730ef-150b-4be7-a00f-82ba6c50792b
```text
23:06:07 FAILED   - JIT/Regression/CLR-x86-JIT/V1-M09.5-PDC/b29068/b29068/b29068.sh
23:06:07                BEGIN EXECUTION
23:06:07                /home/coreclr/bin/tests/Windows_NT.x64.Checked/Tests/coreoverlay/corerun b29068.exe
23:06:07                qemu: Unsupported syscall: 389
23:06:07 
               ./b29068.sh: line 243:   905 Segmentation fault      (core dumped) $_DebuggerFullPath "$CORE_ROOT/corerun" $ExePath $CLRTestExecutionArguments
23:06:07                Expected: 100
23:06:07                Actual: 139
23:06:07                END EXECUTION - FAILED
```

cc @AndyAyersMS

**Comment by benaadams** [comment]
`qemu: Unsupported syscall` is https://github.com/dotnet/coreclr/issues/12972

**Comment by ahsonkhan** [comment]
> Path failures

@JeremyKuhne, @danmosemsft 
I think [this](https://github.com/dotnet/coreclr/pull/16535#issuecomment-368242603) test failure is caused by the recent changes to to PathHelper.Normalize here: https://github.com/dotnet/coreclr/pull/16478 

Is it a mirroring issue - https://github.com/dotnet/coreclr/pull/16447#issuecomment-367907223?
cc @Anipik

**Comment by ahsonkhan** [comment]
> Sounds like we're also missing a test in corefx?

Yes, I think so, We may be missing a test case on the boundary between 0x7F and 0x80 where we aren't calling into the OS when we should. I wonder if the results are any different though.

**Comment by AndyAyersMS** [comment]
I though the plan was to decommission the Tizen tests as they are flaky. @RussKeldorph ??

**Comment by JeremyKuhne** [comment]
@ahsonkhan Go ahead and disable the path test against me and I'll fix it asap- only `~` paths go down this code path. If you see any failures _without_ a `~` in the path let me know. (Note that for some reason the PathTests class isn't executing correctly, which is why I missed this. It appears that after a certain number of test methods tests get skipped silently. Full investigation is ongoing.)

**Comment by RussKeldorph** [comment]
Re: Tizen: The plan is to remove the emulator-based Tizen testing from default PR-triggers, due to the aforementioned flakiness, after we have reliable hardware-based ARM Linux testing in PR, which@BruceForstall is working on.

File: src/mscorlib/shared/System/Globalization/CompareInfo.cs (hunk 4 above)
Comment (ahsonkhan) [reply src/mscorlib/shared/System/Globalization/CompareInfo.cs]: @benaadams, thoughts?

**Comment by ahsonkhan** [comment]
> Go ahead and disable the path test against me and I'll fix it asap- only `~` paths go down this code path. If you see any failures _without_ a `~` in the path let me know. (Note that for some reason the PathTests class isn't executing correctly, which is why I missed this. It appears that after a certain number of test methods tests get skipped silently. Full investigation is ongoing.)

@JeremyKuhne, does this PR resolve the failing tests were are seeing here, or do we still need to disable some tests?
https://github.com/dotnet/corefx/pull/27449

**Comment by ahsonkhan** [comment]
@dotnet-bot test Windows_NT x64 Checked corefx_baseline

**Comment by ahsonkhan** [comment]
CI is passing (modulo Tizen), is this good to merge?

File: src/mscorlib/shared/System/Globalization/CompareInfo.cs (hunk 4 above)
Comment (benaadams) [reply src/mscorlib/shared/System/Globalization/CompareInfo.cs]: @jkotas pointed out it already is https://github.com/dotnet/coreclr/pull/16535#discussion_r170400319

Comment (danmosemsft) [reply src/mscorlib/shared/System/Globalization/CompareInfo.cs]: I was referring to the value, not GlobalizationMode. Ie put either 0xFFFF or 0x7F in a readonly static field. But seems unlikely that would be any faster than a conditional check each time here.
